tests/calloc_test.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/common.h>
+#include <aws/common/math.h>
+#include <aws/testing/aws_test_harness.h>
+
+#ifdef __MACH__
+#    include <CoreFoundation/CoreFoundation.h>
+#endif
+
+AWS_TEST_CASE(test_calloc_fallback, s_test_calloc_fallback_fn)
+static int s_test_calloc_fallback_fn(struct aws_allocator *allocator, void *ctx) {
+    (void)allocator;
+    (void)ctx;
+
+    struct aws_allocator my_alloc = *aws_default_allocator();
+    my_alloc.mem_calloc = NULL;
+
+    /* Check that calloc gives 0ed memory */
```

Comment:
Feels like this logic could be put in a function and invoked from both tests.

Reply:
Done

Reply:
Done